src/Analyzers/MSTest.Analyzers/AssemblyCleanupShouldBeValidAnalyzer.cs

```diff
@@ -61,6 +62,8 @@ public override void Initialize(AnalysisContext context)
         INamedTypeSymbol? valueTaskSymbol, bool canDiscoverInternals)
     {
         var methodSymbol = (IMethodSymbol)context.Symbol;
+        var containingTypeSymbol = context.Symbol.ContainingType;
```

Comment:
I would move this variable close to where it's used. Also, I would probably just drop it and use directly `context.Symbol.ContainingType.IsGenericType` on line 80.